src/Inputs/SearchBarExpandable/SearchBarExpandable.tsx

```diff
@@ -30,16 +30,38 @@ export const SearchBarExpandable: React.FC<SearchBarExpandableProps> = ({
         onInput(event.currentTarget.value);
     };
 
+    const inputRef=useRef<HTMLInputElement>(null)
+
+    const expandSearchBox=useCallback(()=>{
+        state[1](true)
```

Comment:
I know modal does this, but modal is not correct do state, setState. It is less props, but it is also confusing to anyone who doesn't understand this.